src/network/ExtraInfoBuilder.ts

```diff
@@ -65,21 +70,21 @@ export class ExtraInfoBuilder {
       return;
     }
 
-    const index: number | undefined = this.getRequestIndex(req);
+    const index: number = this.getRequestIndex(req);
 
     const requestExtraInfo: RequestExtraInfo | undefined = this
       ._requestExtraInfo[index];
 
     if (requestExtraInfo) {
       req.addExtraRequestInfo(requestExtraInfo);
-      this._requestExtraInfo[index] = null;
+      delete this._requestExtraInfo[index];
```

Comment:
Aren't both needed? `delete` will reclaim the memory, whereas assigning value to `null` will clear the reference.

ditto elsewhere

Reply:
`delete` explicitly makes the property of an object undefined, marks it for removal by GC

Reply:
You're right, should've checked :)